src/com/owncloud/android/operations/RefreshFolderOperation.java

```diff
@@ -349,11 +349,6 @@ private void synchronizeData(ArrayList<Object> folderAndFiles, OwnCloudClient cl
         // get 'fresh data' from the database
         mLocalFolder = mStorageManager.getFileByPath(mLocalFolder.getRemotePath());
 
-        // parse data from remote folder 
-        OCFile remoteFolder = FileStorageUtils.fillOCFile((RemoteFile) folderAndFiles.get(0));
```

Comment:
Because it was creating a new OCFile from the remote file data, without enough information like the favourite state.

I replaced it with an operation from the storage manager (line 440) which gets all the needed data including favorite state, since the folders to sync here are always previously created, to save correctly the folder without losing its data.